packages/react-input/src/components/Input/useInputStyles.ts

```diff
@@ -167,15 +172,20 @@ const useRootStyles = makeStyles({
   },
   disabled: {
     cursor: 'not-allowed',
+    backgroundColor: tokens.colorTransparentBackground,
     ...shorthands.border('1px', 'solid', tokens.colorNeutralStrokeDisabled),
     ...shorthands.borderRadius(tokens.borderRadiusMedium), // because underline doesn't usually have a radius
+    '@media (forced-colors: active)': {
+      ...shorthands.borderColor('GrayText'),
+    },
   },
 });
 
 const useInputElementStyles = makeStyles({
   base: {
     boxSizing: 'border-box',
     flexGrow: 1,
+    minWidth: 0, // required to make the input shrink to fit the wrapper
```

Comment:
What about setting flexShrink?

Reply:
It defaults to 1 (the desired value) but for some reason having a minWidth is also required to make it apply properly 🤷‍♀️

Reply:
it's b/c it appears there's some browser default for minWidth on text-like inputs that isn't exposed (you can verify by making a block input with no other styles, and shrinking the browser width down far enough)